admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyModuleTest.kt

```diff
@@ -0,0 +1,249 @@
+                }
+            }
+        }) {
+            with(handleRequest(HttpMethod.Get, "/api") {
+                addHeader(API_KEY_HEADER, validApiKey)
+            }) {
+                assertEquals(HttpStatusCode.OK, response.status())
+            }
+        }
+    }
+
+    @Test
+    fun `without api key, api-key authenticated request must fail with 401 status`() {
+        withTestApplication({
+            withApiKeyModule {
+                configureMocks()
+            }
+            routing {
+                authenticate("api-key") {
+                    configureGetApiRoute()
+                }
+            }
+        }) {
+            with(handleRequest(HttpMethod.Get, "/api") {
+                //no adding API key header
+            }) {
+                assertEquals(HttpStatusCode.Unauthorized, response.status())
+            }
+        }
+    }
+
+    @Test
+    fun `given invalid api key, api-key authenticated request must fail with 401 status`() {
+        val invalidApiKey = "invalid-api-key"
+        whenever(mockApiKeyBuilder.parse(invalidApiKey)).doReturn(ApiKey(123, "wrong-secret"))
+        wheneverBlocking(mockApiKeyRepository) { findById(123) }.doReturn(createTestApiKeyEntity(apiKeyHash = "hash"))
+        whenever(mockPasswordService.matchPasswords("wrong-secret", "hash")).doReturn(false)
+
+        withTestApplication({
+            withApiKeyModule {
+                configureMocks()
+            }
+            routing {
+                authenticate("api-key") {
+                    configureGetApiRoute()
+                }
+            }
+        }) {
+            with(handleRequest(HttpMethod.Get, "/api") {
+                addHeader(API_KEY_HEADER, invalidApiKey)
+            }) {
+                assertEquals(HttpStatusCode.Unauthorized, response.status())
+            }
+        }
+    }
+
+    @Test
+    fun `given expired api key, api-key authenticated request must fail with 401 status`() {
+        val expiredApiKey = "expired-api-key"
+        whenever(mockApiKeyBuilder.parse(expiredApiKey)).doReturn(ApiKey(123, "key-secret"))
+        wheneverBlocking(mockApiKeyRepository) { findById(123) }
+            .doReturn(createTestApiKeyEntity(expiresAt = LocalDateTime.now().minusDays(1)))
+        whenever(mockPasswordService.matchPasswords(eq("key-secret"), any())).doReturn(true)
+
+        withTestApplication({
+            withApiKeyModule {
+                configureMocks()
+            }
+            routing {
+                authenticate("api-key") {
+                    configureGetApiRoute()
+                }
+            }
+        }) {
+            with(handleRequest(HttpMethod.Get, "/api") {
+                addHeader(API_KEY_HEADER, expiredApiKey)
+            }) {
+                assertEquals(HttpStatusCode.Unauthorized, response.status())
+            }
+        }
+    }
+
+    @Test
+    fun `given api key of blocked user, api-key authenticated request must fail with 403 status`() {
```

Comment:
Is this consistent with jwt and oauth authentication flow behavior for blocked users? (for client API)

Reply:
it is consistent